google/cloud/bigtable/cluster.py

```diff
@@ -323,20 +366,61 @@ def update(self):
 
             before calling :meth:`update`.
 
+            If autoscaling is already enabled, manual scaling will be silently ignored.
+            To disable autoscaling and enable manual scaling, use the :meth:`disable_autoscaling` instead.
+
         :rtype: :class:`Operation`
         :returns: The long-running operation corresponding to the
                   update operation.
         """
         client = self._instance._client
-        # We are passing `None` for third argument location.
-        # Location is set only at the time of creation of a cluster
-        # and can not be changed after cluster has been created.
-        return client.instance_admin_client.update_cluster(
-            request={
-                "serve_nodes": self.serve_nodes,
-                "name": self.name,
-                "location": None,
-            }
+
+        update_mask_pb = field_mask_pb2.FieldMask()
+
+        if self.serve_nodes:
+            update_mask_pb.paths.append("serve_nodes")
+
+        if self.min_serve_nodes or self.max_serve_nodes or self.cpu_utilization_percent:
+            update_mask_pb.paths.append("cluster_config")
```

Comment:
I think we should add the more granular path - `cluster_config.cluster_autoscaling_config` and update `_to_pb()` to reflect that

Reply:
Updated.